### ISSUE TITLE
Unable to search or lookup artist songs

### ISSUE THREAD

**Issue by Saevath** [aome510/spotify-player#617]
**Describe the bug**
Whenever I try to search a song or look up an artist's playlist the player gets stuck, from the logs I can see that the fetch request seems to be failing.

**To Reproduce**
Install spotify-player
Authenticate
Try to search a song title or artist

**Expected behaviour**
View the searched songs

**Log and backtrace**
2024-11-27T21:43:17.059424Z  INFO client_request{request=GetContext(Artist(ArtistId("2d0hyoQ5ynDBnkvAbJKORj")))}: spotify_player::client: Get artist context: spotify:artist:2d0hyoQ5ynDBnkvAbJKORj
2024-11-27T21:43:17.326311Z ERROR client_request{request=GetContext(Artist(ArtistId("2d0hyoQ5ynDBnkvAbJKORj")))}: spotify_player::client::handlers: Failed to handle client request: get related artists: http error: status code 404 Not Found

**Screenshots**
N/A

**Environment**

- OS: Arch Linux
- Application version: 0.20.1
- Application features: pulseaudio-backend,streaming,image,notify

**Additional context**
N/A

**Comment by yawnbo** [comment]
Same issues, this happens when searching
```
2024-11-27T23:31:48.533617Z ERROR client_request{request=Search("blah blah")}: spotify_player::client::handlers: Failed to handle client request: json parse error: invalid type: null, expected struct SimplifiedPlaylist at line 1 column 8664: invalid type: null, expected struct SimplifiedPlaylist at line 1 column 8664

``` 
User playlists still work and appear but new playlists that were made recently aren't appended due to 
```
2024-11-27T23:31:35.293265Z ERROR client_request{request=GetUserPlaylists}: spotify_player::client::handlers: Failed to handle client request: invalid type: null, expected struct SimplifiedPlaylist at line 1 column 4901
```

**Comment by j-danner** [comment]
Might this be related to recent the changes in the web API?

https://developer.spotify.com/blog/2024-11-27-changes-to-the-web-api

**Comment by ociton** [comment]
I also gave the exact same issue

Arch, spotify-player-pipewire from aur

**Comment by aome510** [comment]
Can confirm that I also got many 404 errors. Might be related to the new Spotify API changes. Spotify always breaks their APIs 🤦 . I'll take a look over the weekend to see how to fix this issue.

**Comment by Saevath** [comment]
I can confirm that using the default Spotify client_id has fixed the issue for me.

**Comment by aome510** [comment]
Should be fixed with #618. Can anyone try latest `master` and confirm it's the case?

**Comment by IdoEsports** [comment]
> Should be fixed with #618. Can anyone try latest `master` and confirm it's the case?

Great work! Works perfectly on mac os after building master.

**Comment by ociton** [comment]
when will it be merged into aur package?
